### ISSUE TITLE
VST/AU plugin instances not resizable

### ISSUE THREAD

**Issue by damiensellier** [RomanKubiak/ctrlr#339]
Hi,
That would be so nice to have the option to make the VST container resizeable. Right now the restricted instances are locked by default. Nowadays people use either HD displays or HiDPI 4K+ screens, it's either too big or too small.
Being able to apply zoom factors and resizing the container proportionally would be a nice added value instead of having to export 2 different size instances.

JUCE functions is available in the class [WIndow >>> Resizable Window](https://docs.juce.com/master/classResizableWindow.html)
"ResizableCornerComponent"  (for VST2 and AU) lets the user the possibility to drag the corner to resize the container.
"setResizable(true, true)" enables the resize.
"setResizeLimits(minWidth, minHeight, maxWidth, maxHeight)" sets the min/max values.

Implementing those functions would be really nice!

Thanks

**Comment by bijlevel** [comment]
Yes, I support that request fully!

**Comment by bijlevel** [comment]
I did manage to compile for Mac a CTRLR version with resizable windows for AU/VST3 plugins.

**Comment by dobo365** [comment]
Excellent. I would be glad to have a version able to export VST3 correctly. Is there one?
With 5.3.201 it is VST2.
Thx

**Comment by bijlevel** [comment]
Only for Mac, but not yet fully tested. CTRLR v5.6.23

**Comment by damiensellier** [comment]
Hi @bijlevel that sounds good! Can you please let us know how do you set the window resizable or not? Is it via Lua when the pannel loads or with a checkbox in the global variables? Or something else? thanks

**Comment by bijlevel** [comment]
> Hi @bijlevel that sounds good! Can you please let us know how do you set the window resizable or not? Is it via Lua when the pannel loads or with a checkbox in the global variables? Or something else? thanks

I have explained in discussion #336 how I did it. Try to compile using my guide 😉 to compile the Mac version for CTRLR. Would be a good excercise! OTOH I can upload the app on the website of Godlike Productions, see discussion #278.

**Comment by damiensellier** [comment]
@bijlevel oh yeah I remember, you already gave me the details. Sure , if godlike can host it I'll download it. Have a nice day

**Comment by Godlike-Productions** [comment]
I've uploaded it.

You can find them all here - https://godlike.com.au/index.php?id=361

**Comment by damiensellier** [comment]
@Godlike-Productions I just downloaded the files on your website. Just for the info, what do you mean about "pace compatible"? thanks ;)

**Comment by Godlike-Productions** [comment]
I use iLok (Pace) to secure my Jupiter 8 Plugin, that I sell.  The PACE wrapper requires some space in the header, that by default isn't available in the AU version.  This version has a compile flag allowing this.  It is otherwise identical.

If you want to add iLok support for anti-piracy and registration, then you must use the PACE compatible version.

**Comment by Godlike-Productions** [comment]
Yes, that is correct.  If you get an agreement with PACE, and Derek is OK with it, I can run you through the process or getting ctrlr stuff wrapped, though I'm far from an expert.  PACE has an NDA clause, so can't really discuss the nuts and bolts in a public forum.
